src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -1630,6 +1633,112 @@ export class ResultsViewPageStore {
         }
     });
 
+    /**
+     * Fetch fusion data from the backend API with following request parameters:
+     * - entrez gene ids
+     * - sample molecular identifiers
+     * - molecular profile ids (only when sample molecular identifiers are not provided)
+     * @type {MobxPromiseUnionType<StructuralVariant[]>}
+     */
+    readonly fusions = remoteData<StructuralVariant[]>({
+        await: () => [
+            this.genes,
+            this.selectedMolecularProfiles,
+            this.samples,
+            this.studyIdToStudy
+        ],
+        invoke: async () => {
+            let molecularProfiles: Array<string> = [];
+            let samples: SampleMolecularIdentifier[] = [];
+            let params: StructuralVariantFilterExt = {entrezGeneIds:[]};
+            if (this.genes.result) {
+                params.entrezGeneIds = _.map(this.genes.result, (gene: Gene) => gene.entrezGeneId);
+            }
+            // compose request parameters
+            if (this.samples.result) {
+                // get sample molecular identifiers
+                samples = this.samples.result.map(sample => {
+                    return {
+                        molecularProfileId: sample.studyId + '_structural_variants',
+                        sampleId: sample.sampleId
+                    };
+                });
+                params.sampleMolecularIdentifiers = samples;
+            } else {
+                // get molecular profile ids
+                if (this.selectedMolecularProfiles.result) {
+                    molecularProfiles = this.selectedMolecularProfiles.result.filter(profile => {
+                        return profile.molecularAlterationType === AlterationTypeConstants.STRUCTURAL_VARIANT;
+                    }).map(profile => {
+                        return profile.molecularProfileId;
+                    });
+                    params.molecularProfileIds = molecularProfiles;
+                }
+            }
+            return await client.fetchStructuralVariantsUsingPOST({'structuralVariantFilter':<StructuralVariantFilter> params});
+        }
+    });
+
+    /**
```

Comment:
these interface comments seem redundant given that the typescript annotation says the same thing.  are the comments being auto-generated?  textual comments that explain the function are great though.

Reply:
I agree with @alisman I personally think it's harder to read and clutters the code and it creates maintenance burden, mostly duplicates work that should properly be done by the param names and type names. Any further function documentation can be put in a standard comment inside the body of the function

Reply:
I think we added it to have an explicit definition of every function, I don't think that the comments were auto-generated. I will remove them because indeed with the function names and the types of the parameters you can easily understand what is going on.